File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/TypeTemplateService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbTypeTemplate;							  
import entity.PageResult;
import entity.Result;
/**
 * typeTemplate服务层接口
 * @author jieweili
 *
 */
public interface TypeTemplateService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbTypeTemplate> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param typeTemplate
     * @return
     */
    public Result saveTypeTemplate(TbTypeTemplate typeTemplate);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbTypeTemplate getTypeTemplateById(Long id);

    /**
     * 修改对象
     * @param typeTemplate
     * @return
     */
    Result updateTypeTemplate(TbTypeTemplate typeTemplate);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeTypeTemplate(Long[] idList);

    /**
     * 根据条件分页查询
     * @param typeTemplate
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listTypeTemplateByExample(TbTypeTemplate typeTemplate, Integer pageNum, Integer pageSize);
 
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/typeTemplateService.js
app.service('typeTemplateService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../typeTemplate/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../typeTemplate/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeTypeTemplateList = function (idList) {
        return $http.get('../typeTemplate/removeTypeTemplate.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../typeTemplate/findOne.do?id=' + id)
    };

    //修改
    this.updateTypeTemplate = function (typeTemplate) {
        return $http.post('../typeTemplate/updateTypeTemplate.do', typeTemplate)
    };

    //新增
    this.saveTypeTemplate = function (typeTemplate) {
        return $http.post('../typeTemplate/saveTypeTemplate.do', typeTemplate)
    };


});
